### ISSUE TITLE
Special Characters

### ISSUE THREAD

**Issue by DogDaysA** [bastimeyer/dayz-linux-cli-launcher#3]
There are a few mods out there that use [ ] and the  + sign.  Is it possible to get these added as well?  I am not sure if it's just as easy to go with the number of the mod instead and trying to use the name.  We used ping perfect in the passed and they used numbers instead of the name, considering there is no naming standard.

**Comment by bastimeyer** [comment]
Can you share a link to one of these mods? Symlinks with these characters shouldn't cause any issues as these are all allowed and supported characters on any filesystem.

The reason why I went with mod names instead of mod IDs is that the official launcher is using mod names as well (or at least has been using - not sure if this has changed). Using IDs makes it rather difficult for humans if you want to know which mods are currently installed and which are not. Mod names don't have this problem.

The only special character which gets replaced is currently `'` due to a limitation of Steam's game launch parameters, which doesn't properly escape the inputs.
https://github.com/bastimeyer/dayz-linux-cli-launcher/blob/0.1.2/dayz-launcher.sh#L232

**Comment by bastimeyer** [comment]
Ok, looks like Steam's launcher script which handles the game parameters doesn't work correctly:
```sh
+ steam -applaunch 221100 '-mod=@[Remastered] Arma Weapon Pack'
/media/storage/SteamHome/.local/share/Steam/steam.sh: line 574: no match: -mod=@[Remastered]
```

So there are three solutions which come to my mind

1. Use mod IDs instead of mod names
2. Replace any non-alphanumerical characters
3. Do both of the above: `${id}-${name}`

3 seems to be the best solution, in case there are mod name collisions. I'll have a look a this later.

**Comment by DogDaysA** [comment]
Take your time!  I just dropped you an email.  Thank you again!!!

**Comment by bastimeyer** [comment]
Fixed by 4feae2ca32421252ad68680e6555fb307a5b236c
https://github.com/bastimeyer/dayz-linux-cli-launcher/blob/0.2.0/CHANGELOG.md

> I just dropped you an email

Thanks. No need to give me your server address, should be working fine now, regardless of the names of the mods being used.

Also thanks for trying to resolve the server API issues. I've already commented on this (I believe on reddit), and ideally I wouldn't have to use a third party JSON API for querying server IPs, but that's a bit difficult to achieve via BASH. If I had chosen Python for example, this would've been quite simple, but writing the launcher in BASH was the quickest solition. Python would've also allowed for creating a simple GUI on top of the CLI. Maybe that's a project for another day.
